### PR TITLE
vCard v4 Support

### PR DESCRIPTION
**Pull request by ksmurchison**
Beginnings of #3715 .  Conversion functions need a lot more work

### CODE REVIEW

**Comment by dilyanpalauzov** [comment]
The text at https://www.cyrusimap.org/3.4/imap/rfc-support.html for “RFC 6352 CardDAV: vCard Extensions to WebDAV” needs adjustment.  Currently it says:
> Cyrus IMAP accepts over CardDAV both vCard 3.0 and vCard 4.0, but does not advertise over CARDDAV:supported-address-data vCard 4.0 support, announces only vCard 3.0. Moreover, Cyrus IMAP does not convert between vCard 3.0 and vCard 4.0. It is applications’ responsibility not to misinterpret the vCard version and damage the data.

On the same page, the hints after RFC 6350, RFC 2425, and RFC 2426 reference the above comment.

File: imap/http_jmap.c [hunk 1]
```diff
@@ -553,6 +553,8 @@ static char *parse_accept_header(const char **hdr)
         struct accept *tmp;
         for (tmp = accept; tmp && tmp->token; tmp++) {
             free(tmp->token);
+            free(tmp->version);
+            free(tmp->charset);
         }
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/http_jmap.c:556]: Done

**Review by rsto — APPROVED** [review]
Looks good to me.

**Review by rsto — APPROVED** [review]
Looks good to me

**Pull request merged**